root/static/scripts/tests/Control/URLCleanup.js

```diff
@@ -4848,6 +5405,11 @@ testData.forEach(function (subtest, i) {
       }
       tested = true;
     }
+    if (subtest.expected_favicon_class) {
+      const actualFaviconClass = getFaviconClass(actualCleanUrl);
+      st.equal(actualFaviconClass, subtest.expected_favicon_class, 'Correct favicon');
+      tested = true;
```

Comment:
Does that imply that a test can just consist of `input_url` and `expected_favicon_class` ?
If it is tested against the clean URL, `expected_clean_url` should be mandatory.
Maybe the documentation about test properties in dcabded48be065374c55bfb789ad588daa363729 commit’s message should be added to this file?

Reply:
Yes, IMO that should be ok, shouldn't it? It's tested against the return of cleanURL, which is the original if there's no cleanup function, so effectively, it is tested against whatever will actually be added to the DB.

Reply:
Oh, or is your idea (seeing the miura.com example above) that there should be an `expected_clean_url` even if the expectation is there's no cleanup and you get the same that came in?

Reply:
> Yes, IMO that should be ok, shouldn't it?

Not sure what “that” and “it” refer to.

> It's tested against the return of `cleanURL(), which is the original if there's no cleanup function`

It is hiding the URL being actually tested for favicon class, from both the test file and the test output.

The clean URL may significantly differ from the input URL, that’s why I suggested to make `expected_clean_url` mandatory for use with `expected_favicon_class`.

> so effectively, it is tested against whatever will actually be added to the DB.

So effectively, it doesn’t test against actual unclean URLs already in the DB.
Let’s use the above thread https://github.com/metabrainz/musicbrainz-server/pull/2256#discussion_r709123871 for follow-up on this point.

Reply:
> Oh, or is your idea (seeing the miura.com example above) that there should be an `expected_clean_url` even if the expectation is there's no cleanup and you get the same that came in?

Yes. (Did post the previous comment before seeing that one.)

Reply:
Ok, we can always do that and always have `expected_clean_url` - I just thought that was a bit misleading because it suggested a cleanup was being run, but maybe not :)